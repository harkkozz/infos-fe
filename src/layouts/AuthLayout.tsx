import React, { PropsWithChildren } from 'react';

import HomeIcon from '@iconscout/react-unicons/icons/uil-home';
import { Button } from 'antd';
import { useNavigate } from 'react-router';

import styles from './AuthLayout.module.scss';

const AuthLayout: React.FC<PropsWithChildren> = ({ children }) => {
  const navigate = useNavigate();

  return (
    <>
      <div className={styles.homeButtonWrapper}>
        <Button type="link" onClick={() => navigate({ pathname: '/' })}>
          <HomeIcon size="30" color="#1677be" />
        </Button>
      </div>

      <div className={styles.auth}>
        <div className={styles.formWrapper}>{children}</div>
      </div>
    </>
  );
};

export default AuthLayout;
