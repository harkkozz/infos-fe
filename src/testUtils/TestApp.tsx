import React, { PropsWithChildren } from 'react';

import { MemoryRouter } from 'react-router-dom';

interface Props {
  path?: string;
}

const Test: React.FC<PropsWithChildren<Props>> = ({ path, children }) => {
  return <MemoryRouter initialEntries={[!path ? '/' : path]}>{children}</MemoryRouter>;
};

export default Test;
