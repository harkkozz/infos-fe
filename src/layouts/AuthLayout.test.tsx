import React from 'react';

import { render } from '@testing-library/react';
import TestApp from 'testUtils/TestApp';
import { describe, expect, it } from 'vitest';

import AuthLayout from './AuthLayout';

describe('Auth Layout', () => {
  it('should match snapshot', () => {
    const { baseElement } = render(
      <TestApp>
        <AuthLayout />
      </TestApp>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
