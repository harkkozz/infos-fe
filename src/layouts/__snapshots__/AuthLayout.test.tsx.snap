// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Auth Layout > should match snapshot 1`] = `
<body>
  <div>
    <div
      class="_homeButtonWrapper_df52fe"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-1vtf12y ant-btn-link"
        type="button"
      >
        <svg
          fill="#1677be"
          height="30"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.66,10.25l-9-8a1,1,0,0,0-1.32,0l-9,8a1,1,0,0,0-.27,1.11A1,1,0,0,0,3,12H4v9a1,1,0,0,0,1,1H19a1,1,0,0,0,1-1V12h1a1,1,0,0,0,.93-.64A1,1,0,0,0,21.66,10.25ZM13,20H11V17a1,1,0,0,1,2,0Zm5,0H15V17a3,3,0,0,0-6,0v3H6V12H18ZM5.63,10,12,4.34,18.37,10Z"
          />
        </svg>
      </button>
    </div>
    <div
      class="_auth_df52fe"
    >
      <div
        class="_formWrapper_df52fe"
      />
    </div>
  </div>
</body>
`;
